### ISSUE TITLE
'httpserver' module included by default even when specific module(s) list specified in 'make' arguments

### ISSUE THREAD

**Issue by lgoldstein** [cloudius-systems/osv#390]
This is a discussion that started on some other issue that discovered this due to a bug, but I thought it appropriate to open as a separate issue. Basically, the principle at stake here is "what modules should be included in an image when a _make_ command is issued ?".

The current (latest) image build automatically adds the _default_ modules specified in the _config.json_ file even if the user specified the **exact** modules via _make image=X,Y,Z_. In other words, the _default_ modules are **added** to the ones specified by the user - unless **specifically excluded** by the user via _make image=X,Y,Z,-M1,-M2_. According to _Or Cohen_'s remark: _c1c6054 introduced this. The reasoning is that the API server is an essential part of OSv. According to the patch, it can be removed with: "make image=XXX,-httpserver"_.

My claim is that this is not a desired behavior and furthermore it conflicts with the _simple, lean and mean O/S_ vision that Dor and Nadav agreed with during a very fruitful chat we had. My reasoning is as follows (I am summarizing what I posted on the #384 so please read there for more justification): 

If the user chooses to build the **default** image (i.e., just _make_), then it is OK to bundle whatever the developers feel are "best practices" or "recommended modules". However, if the user chooses to select only certain modules, then **only** those modules + the **minimum** dependent ones should be included. Only these modules are truly **essential** - e.g., IMO "httpserver" (the trigger for this issue) is definitely **not** "essential" since I can run my image without it and have it work perfectly and smoothly. Such a module may be "(highly) recommended", "extremely useful" (choose your adjective) - but **not** "essential".

IMO, this behavior actually actually **tarnishes** the principle of _simplicity_ which I believe lies at the core of OSV since it requires "opting out" instead of "opting in". Imagine that developers decide that Ruby, Python, MySQL, etc. are also "essential" (whereas we can all agree that they are not...) - then every time I build my image I need to consult some obscure file to see what else was decided "essential" when de-facto it isn't. I urge you **not** to go on the path of the "legacy" systems that keep bundling more and more modules they call "essential" - creating bloated O/S that are very hard to slim down. Keep OSV simple, lean and mean as its creators intended it.

In summary, I would like to suggest some solution - instead of using a "static" file that specifies the configuration (_config.json_) rename it _default-config.json_ and allow the user to specify _my-config.json_ "override" to be used when building the image - e.g., _make config=my-config.json_.
